app/main.py

```diff
@@ -1,3 +1,52 @@
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: int | float) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | float) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> "Distance":
+        return Distance(round(self.km / other, 2))
```

Comment:
There is a potential division by zero error in the `__truediv__` method. It's important to check if `other` is zero before performing the division to avoid runtime errors. Consider adding a condition to handle this case.